lib/rspec-puppet/support.rb

```diff
@@ -402,10 +415,15 @@ def build_catalog_without_cache(nodename, facts_val, trusted_facts_val, hiera_co
 
       node_obj = Puppet::Node.new(nodename, { :parameters => node_params, :facts => node_facts })
 
+      trusted_info = ['remote', nodename, trusted_facts_val]
+      if Puppet::Util::Package.versioncmp(Puppet.version, '6.14.0') >= 0
+        trusted_external_data = args[-1]
```

Comment:
This here is more concerning. The args splat's main use was it's side-effect to act as a cache key. This now hard-codes "the last element is the trusted_external data", which is tenuous at best.

Considering that the original use of the build_catalog splat was it's use as cache key, there's the theoretical chance that someone somewhere has (mis-)used this API for their own reasons.

Being saddled with having to support everything back to the stone-age, there's no really good solutions here. One idea would be to start towards cleaning this up for a 3.0 release and change it something like this:

```ruby
def build_catalog_without_cache(nodename, facts_val, trusted_facts_val, hiera_config_val, code, exported, node_params, *_)
  build_catalog_without_cache_new({
    nodename: nodename,
    facts_val: facts_val,
    trusted_facts_val: trusted_facts_val,
    hiera_config_val: hiera_config_val,
    code: code,
    exported: exported,
    node_params: node_params,
    trusted_external: {},
  })
end

def build_catalog(*args)
  @@cache.get(*args) do |*args|
    if args.length == 1
      build_catalog_without_cache_new(args.first)
    else
      build_catalog_without_cache(*args)
    end
  end
end

def load_catalogue(type, exported = false, manifest_opts = {})
  # ...
  catalogue = build_catalog({
    nodename: node_name,
    facts_val: build_facts,
    trusted_facts_val: trusted_facts_hash(node_name),
    hiera_config_val: hiera_config_value,
    code: build_code(type, manifest_opts),
    exported: exported,
    node_params: node_params_hash,
    trusted_external: {},
    ignored_cache_params: {
      hiera_data_value: hiera_data_value,
      trusted_server_facts: RSpec.configuration.trusted_server_facts
    }
  })
  # ...
```

This adds a new 1-argument overload for `build_catalog` that should be a net-new API addition (and thus OK for a .y release) and a net-new `build_catalog_without_cache_new` method that can use the has (and later kwargs) for more flexibility without painting us further into a corner.

Reply:
Updated a while ago but am wondering why Travis is taking so long...

Reply:
Thanks for the quick response!

re travis: That's what acquisitions do to services :-(